src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -692,95 +705,42 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
-
-								//Likely a SQL Injection. Raise it
-								String extraInfo = null;
-								if (strippedOutput[booleanStrippedUnstrippedIndex]) {
-									extraInfo = Constant.messages.getString(MESSAGE_PREFIX + "alert.expressionbased.extrainfo", modifiedParamValue, "");
-								} else {
-									extraInfo = Constant.messages.getString(MESSAGE_PREFIX + "alert.expressionbased.extrainfo", modifiedParamValue, "NOT ");
-								}
-
-								//raise the alert, and save the attack string for the "Authentication Bypass" alert, if necessary
-								sqlInjectionAttack = modifiedParamValue;
-								bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, getName(), getDescription(),
-										null, //url
-										param, sqlInjectionAttack,
-										extraInfo, getSolution(), "", msg4);
-
-								sqlInjectionFoundForUrl = true;
-							}
+					// This check is implemented using two variant PLUS(+) and MULT(*)
+					try{
+						// PLUS variant check the param value "3-2" gives same result as original request and param value "4-2" gives different result if original param value is 1
+						//set the parameter value to a string value like "3-2", if the original parameter value was "1"
+						int paramPlusTwo = addWithOverflowCheck(paramAsInt, 2);
+						String modifiedParamValueForAdd = String.valueOf(paramPlusTwo) + "-2";
+						//set the parameter value to a string value like "4-2", if the original parameter value was "1"
+						int paramPlusThree = addWithOverflowCheck(paramAsInt, 3);
+						String modifiedParamValueConfirmForAdd = String.valueOf(paramPlusThree) + "-2";
+						//Do the attack for ADD variant
+						expressionBasedAttack(param, modifiedParamValueForAdd, modifiedParamValueConfirmForAdd);
+						//bale out if we were asked nicely
+						if (isStop()) { 
+							log.debug("Stopping the scan due to a user request");
+							return;
 						}
+						// MULT variant check the param value "2/2" gives same result as original request and param value "4/2" gives different result if original param value is 1
+						//set the parameter value to a string value like "2/2", if the original parameter value was "1"
+						int paramMultTwo = multiplyWithOverflowCheck(paramAsInt, 2);
+						String modifiedParamValueForMult = String.valueOf(paramMultTwo) + "/2";
+						//set the parameter value to a string value like "4/2", if the original parameter value was "1"
+						int paramMultFour = multiplyWithOverflowCheck(paramAsInt, 4);
+						String modifiedParamValueConfirmForMult = String.valueOf(paramMultFour) + "/2";
+						//Do the attack for MULT variant
+						expressionBasedAttack(param, modifiedParamValueForMult, modifiedParamValueConfirmForMult);
 						//bale out if we were asked nicely
 						if (isStop()) { 
 							log.debug("Stopping the scan due to a user request");
 							return;
 						}
+					} catch (Exception e){
+						if (this.debugEnabled) {
+							log.debug("The parameter value [" + origParamValue + "] is TOO large to test this check");
```

Comment:
Since you also caught underrun (thanks!) this message no longer works generically, perhaps:
`log.debug(""Caught " + ex.getClass().getName() + " " + ex.getMessage() + "When performing integer math with the parameter value [" + origParamValue + "]");`